e2e-tests/src/test/eth_to_azero.rs

```diff
@@ -0,0 +1,111 @@
+    let weth_eth_address = eth_contract_addresses.weth9.parse::<Address>()?;
+
+    let weth_abi = eth::contract_abi(&config.contract_metadata_paths.eth_weth9)?;
+    let weth = eth::contract_from_deployed(weth_eth_address, weth_abi, &eth_signed_connection)?;
+
+    let transfer_amount = utils::parse_ether(config.test_args.transfer_amount)?;
+    let send_receipt = eth::send_tx(
+        eth_account_address,
+        weth_eth_address,
+        transfer_amount + 100,
+        &eth_signed_connection,
+    )
+    .await?;
+    info!("Send tx receipt: {:?}", send_receipt);
+
+    let most_address = eth_contract_addresses.most.parse::<Address>()?;
+
+    let approve_args = (
+        most_address,
+        utils::parse_ether(config.test_args.transfer_amount)?,
+    );
+
+    let approve_receipt =
+        eth::call_contract_method(weth, "approve", config.eth_gas_limit, approve_args).await?;
+    info!("'Approve' tx receipt: {:?}", approve_receipt);
+
+    let most_abi = eth::contract_abi(&config.contract_metadata_paths.eth_most)?;
+    let most = eth::contract_from_deployed(most_address, most_abi, &eth_signed_connection)?;
+
+    let mut weth_eth_address_bytes = [0_u8; 32];
+    weth_eth_address_bytes[12..].copy_from_slice(weth_eth_address.as_fixed_bytes());
+
+    let azero_contract_addresses =
+        azero::contract_addresses(&config.azero_contract_addresses_path)?;
+    let weth_azero_account_id = AccountId32::from_str(&azero_contract_addresses.weth)
+        .map_err(|e| anyhow::anyhow!("Cannot parse account id from string: {:?}", e))?;
+
+    let azero_account_keypair = keypair_from_string("//Alice");
+    let azero_account_address_bytes: [u8; 32] =
+        (*azero_account_keypair.account_id()).clone().into();
+
+    let send_request_args = (
+        weth_eth_address_bytes,
+        utils::parse_ether(config.test_args.transfer_amount)?,
+        azero_account_address_bytes,
+    );
+    let send_request_receipt =
+        eth::call_contract_method(most, "sendRequest", config.eth_gas_limit, send_request_args)
+            .await?;
+    info!("'sendRequest' tx receipt: {:?}", send_request_receipt);
+
+    info!(
+        "Waiting {:?} minutes for finalization",
+        config.test_args.wait_minutes
+    );
+    tokio::time::sleep(tokio::time::Duration::from_secs(
+        60_u64 * config.test_args.wait_minutes,
+    ))
+    .await;
+
+    let azero_connection = azero::connection(&config.azero_node_ws).await;
+
+    let weth_azero_contract = ContractInstance::new(
+        weth_azero_account_id,
+        &config.contract_metadata_paths.azero_token,
+    )?;
+
+    let balance_post_transfer: u128 = weth_azero_contract
+        .contract_read(
+            &azero_connection,
+            "PSP22::balance_of",
+            &[(*azero_account_keypair.account_id()).clone().to_string()],
+        )
+        .await?;
+    info!("balance post transfer: {:?}", balance_post_transfer);
+    assert_eq!(transfer_amount, balance_post_transfer.into());
```

Comment:
I would propose to check difference in balance before and after bridging, this way we might rerun the test without tearing down the network.